src/axom/core/ArrayBase.hpp

```diff
@@ -767,7 +767,7 @@ struct ArrayOpsBase<T, true>
   {
     // Similar to fill(), except we can allocate stack memory and placement-new
     // the object with a move constructor.
-    std::aligned_storage<sizeof(T), alignof(T)> host_buf;
+    alignas(T) unsigned char host_buf[sizeof(T)];
```

Comment:
Minor:
Would it make sense to use a fixed width type, e.g.
```suggestion
    alignas(T) axom::uint8 host_buf[sizeof(T)];
```